### ISSUE TITLE
@sig at steady_state_imposed

### ISSUE THREAD

**Issue by cbarreva** [jmaih/RISE_toolbox#27]
Hi Junior.

I'm currently working on a Regime-Switching model with endogenous transition probabilities. However, I have a problem because the endogenous variable at transition probabilities does not have an unique steady state. So, in order to solve the model I must imposed the steady state (which is the steady state of the second regime). My problem is that I have @sig constants in the policy functions of the second regime, which represents a difference between the SS I am imposing and the SS that RISE would calculate without "'steady_state_imposed', true". I do not understand why those @sig constants appear. When I use exogenous probabilities and "'steady_state_imposed', true" the same problem arises. Am I misunderstanding @sig? Shouldn't appear when the imposed steady state is equal to the steady state of regime 2?

Thank you for your attention.

Sincelery,

Leonardo

**Comment by jmaih** [comment]
Hi Leonardo,

As you have realized, RISE is allergic to cases where the variables entering the transition probabilities do not have a unique steady state. The intuition should be clear: The nonlinear model will be approximated around the steady state. If then, the variables entering the transition probabilities have multiple steady states, the transition probabilities will not be unique. What that means is that we will not know how to form expectations because for a given state of the economy, we have different probabilities for a given event to occur.

Now, when you impose a steady state you do resolve the confusion above but at the same time you create a new set of issues. Imposing a steady state implies that the residuals evaluated in a given regime might be nonzero. But RISE still needs to find a solution to the problem such that for the dynamic model we still have `E[f(x_{t+1},x_t,x_{t-1},e_t)]=0` even if for the static model `f(x_t,x_t,x_t,0)` is different from zero. 

Loosely speaking, RISE approximates the system around a point that is not the steady state. RISE can only achieve this through solving for a constant that compensates for the fact that the regime-specific steady state residuals are different from zero: This is the role of @sig, which then appears already at a first-order approximation because in that case certainty equivalence does not hold. You can easily check this by running exogenous probabilities without imposing the steady state.

Summing up, if you want to use endogenous probabilities, you will have to ensure the variables entering the time-varying transition probabilities have a unique steady state. But this is mostly a weakness of the solution approach rather than a fundamental property of the problem: if we knew how to solve the problem exactly, we would not need to approximate it and hence we would not need to refer to the steady state.

Cheers,

Junior

**Comment by cbarreva** [comment]
Hi Junior,

Thanks a lot for answered me. I am currently trying to solve an occasionally binding constraint model using RISE, and because of that, I need to use endogenous transition probabilities with variables that does not have an unique steady state. So, I need to impose the steady state. I understood the function of @sig constants, but I do not know how to use them in order to replicate the regime-specific steady state. Should I multiply those constants by the policy functions coefficients when trying to obtain the imposed steady state? I need to know how to treat @sig constants.

Thank you for your attention.

Sincelery,

Leonardo

**Comment by jmaih** [comment]
Hi Leonardo,

1. The @sig constants should not be touched, they are part of the solution of the system. 

1.  The variables entering the transition probabilities cannot have multiple values at their steady state as I tried to explained last time. In the same way, the parameters entering the transition probability functions cannot switch. If either the variables or the parameters or both were to assume different values depending on the regime, there would be multiple switching probabilities from one particular state to another,

1. Some authors are uncomfortable with the idea of multiple steady states. But the fact of the matter is that if the system persists in one particular regime for years it is difficult to make the argument that we should entertain the notion of a unique attractor or steady state.

1. My guess is that any problem in which the variables entering the transition probabilities can have different values in each steady state is not well posed. But I could be wrong. Maybe you could share more details about the problem you would like to solve?

Cheers,

Junior

**Comment by cbarreva** [comment]
Hi Junior.

Thank you again for your time. I sent you an email with more details about my problem, and my RISE code.

Sincelery,

Leonardo

**Comment by jmaih** [comment]
Hi Leonardo,

I hope you are making progress after the exchanges we've had.

Cheers,

Junior

**Comment by jkrenz** [comment]
Hi, 

I have a follow up question on this. Im using the "**steady state unique**" option for a model with endogenous transition probabilities. The model has the property that, if we would treat each regime as a different model, the steady state of each model would be quite different from the other. 

Now, I have some questions related to this: 

1) What is the value wich RISE reports as steady state in this case? Is it the ergodic mean?

2) When using the "steady state unique" option, each variable has the same steady state value in both regimes, but the values given under @sig are different in each regime. Now, in order to see which is the point in which the residuals evaluated in a given regime are nonzero, can I just add the value given under @sig to the steady state value?

3) Where in the RISE output can I exactly find the @sig values? I searched for quite some time now, but I didn't find the place where they are stored. 

4) I'm still not sure how to think about the ergodic mean. When I understand correctly RISE computes a theoretical ergodic mean. But could I obtain the ergodic mean by simulating the model for very long, with shock and regime switches, and then just take the mean of each variable over the whole simulation?

Thanks in advance an best regards, 
Johanna

**Comment by jmaih** [comment]
Hi Johanna,


   1. ergodic mean
   2. Irrespective of the point of approximation chosen, RISE tries to
   ensure that Ef(x) =0. Because the point of approximation is not the steady
   state, @sig accounts for the constant terms added so that in the end Ef(x)=0
   3. The @sig values are stored in a column of the matrix of policy
   function Tz containing the coefficients on lag values of x, @sig, and the
   shocks. This means that the column with @sig coefficients is found just
   before those of the shocks.
   4. There is a long-run mean for the transition probabilities. The unique
   steady state is computed as the weighted average of regime-specific means.
   This is the ergodic mean. Your intuition is correct in the sense that in
   theory one could simply simulate the model for a long time and compute the
   mean of the simulation. In practice I don't know how fast convergence will
   occur. This may be problem-specific.

Cheers,

J.

**Comment by jkrenz** [comment]
Thank you very much! That was very helpfull!
